### DISCUSSION TITLE
Setting indentation type for clangd lsp formatting

### DISCUSSION THREAD

**Discussion by nicolasdumitru** [VonHeikemen/lsp-zero.nvim#327] [Q&A] [answered]
Hello! I am trying to get my lsp to format c and c++ code with tabs instead of spaces when using `vim.lsp.buf.format`, but I don't know how to do this and I need some help.
[Here](https://github.com/nicolasdumitru/neovim-configuration/blob/5b0fe4f974a6e0ec37d6b18906ff7d6d098294f5/lua/nick/plugin/spec/lsp-zero.lua) you can find my entire lsp-zero configuration.

**Comment by VonHeikemen** [comment, marked as answer]
I believe you can configure `clangd` formatting options using a file called `.clang-format` in the root of your project.

**Reply by nicolasdumitru** [reply]
That's good to know, but I'd like to set a kind of "base" setting for my projects and then (maybe) override it with `.clang-format` as needed.

**Reply by surbhitt** [reply]
Hey @nicolasdumitru did you figure it out? I wanna do the same thing.

**Reply by nicolasdumitru** [reply]
Yes. First of all @VonHeikemen's answer helped me a lot. What I found is that you can have both what I said (a general default) and a project-specific set of rules. For the general defaults, place a .clang-format with your preferences in your `$HOME` and for the project-specific formatting rules, place a .clang-format in your project's root directory. From what I've tested, clang-format seems to just search the parent directory of the file currently being edited and the its parent directory and so on until it reaches `/`. I don't know if the .clang-format in your project's directory just overrides what is in `$HOME/.clang-format` or it just makes clang-format complete. Also, I haven't tested if git repositories have an impact in determining what .clang-format file is used. This should be a pretty decent starting point. Play around with it. Have fun and I hope this helps you!

My setup (where I've tested this stuff):
OS: Arch Linux (distro shouldn't matter too much/at all I think)
Editor: Neovim v0.9.5 [(setup at the moment of testing this - plugins were up to date at the moment of writing)](https://github.com/nicolasdumitru/neovim-configuration/tree/6c6c42fe44859ca2b227e621a7d7ef0642e79d3e)
clang, clangd, clang-format version: 16.0.6

**Reply by surbhitt** [reply]
thanks a lot

**Reply by nicolasdumitru** [reply]
You're welcome. I hope it's what you were looking for.
